plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/model/HudiReplaceCommitMetadata.java

```diff
@@ -13,28 +13,34 @@
  */
 package io.trino.plugin.hudi.model;
 
+import com.fasterxml.jackson.annotation.JsonCreator;
```

Comment:
> The Hudi replace commit metadata model object was also patched to
deserialize Hudi metadata properly. This was needed so that Trino
users (and the tests for the active timeline) do not see duplicated
records for replace commits.

Where's the test about this? This change is unrelated to fix instant file name path, right? Can we separate a PR?

Reply:
This is tested by the test suite `testReadCopyOnWriteTableWithReplaceCommits` within `TestHudiSparkCompatibility`. If these changes are not applied, clients will see duplicated records before replace commits are cleaned up from the Hudi timeline, so instead of using this assertion in the test:
```java
onHudi().executeQuery("INSERT INTO default." + tableName + " VALUES (1, 'a1', 1000), (2, 'a2', 2000)");
assertThat(onTrino().executeQuery("SELECT id, name FROM hudi.default." + tableName))
                    .containsOnly(
                            row(1, "a1"),
                            row(2, "a2"));
```
We should use this instead (even though we have inserted only 2 records into the table): 
```
onHudi().executeQuery("INSERT INTO default." + tableName + " VALUES (1, 'a1', 1000), (2, 'a2', 2000)");
assertThat(onTrino().executeQuery("SELECT id, name FROM hudi.default." + tableName))
                    .containsOnly(
                            row(1, "a1"),
                            row(2, "a2"),
                            row(1, "a1"),
                            row(2, "a2"));
```
Which looks very strange and I think that these faults are slightly related. Their source is basically next to each other. If you take a look at `HudiTimeline.resetFileGroupsReplaced`:
```java
HudiReplaceCommitMetadata replaceMetadata = HudiReplaceCommitMetadata.fromBytes(
                                metaClient.getActiveTimeline().getInstantDetails(instant).get(),
                                OBJECT_MAPPER,
                                HudiReplaceCommitMetadata.class);
```
- The first fault is that `getInstantDetails` could not create a `Location` instance. That has been fixed in `HudiActiveTimeline`. 
- The second fault is that since the `HudiReplaceCommitMetadata.fromBytes` cannot properly deserialize the class, the list of `partitionToReplaceFileIds` will be empty and clients will see duplicate content.

If you insist, however, we can separate this into two different PRs, that's why I wanted to start the discussion above regarding this in https://github.com/trinodb/trino/pull/18213#issuecomment-1681230801 (see the second paragraph). If that is the case, I will need to change back the tests to this weird form (see above) for the time being (will be fixed in the consecutive PR though). Please let me know what do you think about this.